src/posix/platform/daemon.cpp

```diff
@@ -74,15 +74,21 @@ void GetFilename(Filename &aFilename, const char *aPattern)
 
 int Daemon::OutputFormatV(const char *aFormat, va_list aArguments)
 {
-    char buf[OPENTHREAD_CONFIG_CLI_MAX_LINE_LENGTH + 1];
+    static constexpr char truncatedMsg[] = "(truncated ...)";
+    static_assert(sizeof(OPENTHREAD_CONFIG_CLI_MAX_LINE_LENGTH) < sizeof(truncatedMsg),
```

Comment:
nit: I think it might be neater if the assert is placed below all the variable declarations.